pkgs/tools/archivers/dumpnar/default.nix

```diff
@@ -0,0 +1,26 @@
+{ lib, stdenv, fetchFromGitHub }:
+
+stdenv.mkDerivation rec {
+  pname = "dumpnar";
+  version = "20130101";
+
+  src = fetchFromGitHub {
+    owner = "stephank";
+    repo = "dumpnar";
+    rev = "7b05e204264183532e8592ad132f74ddf05bc428";
+    hash = "sha256-762vgCn2d6QoXRBjpe/SyHKgyQJAV0sEl1prcyf/ClE=";
+  };
+
+  installPhase = ''
+    mkdir -p $out/bin
+    cp dumpnar $out/bin/
+  '';
+
+  meta = with lib; {
+    homepage = "https://github.com/stephank/dumpnar";
+    description = "Minimal tool to produce a Nix NAR archive.";
+    license = licenses.lgpl2Plus;
+    platforms = platforms.all;
+    maintainers = [ maintainers.stephank ];
+  };
```

Comment:
This is code extracted from Nix itself. Trying to use Nix itself here was trouble. It significantly increases the built closure, plus it apparently looks for config and errors even for simple things like creating a NAR. I didn't investigate beyond that point and simply extracted the code. (I had another use for it myself.)